### ISSUE TITLE
Error when --large-index is not specified

### ISSUE THREAD

**Issue by nigiord** [BenLangmead/bowtie2#245]
Hi there,

According to the bowtie2 documentation: http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#small-and-large-indexes

> bowtie2-build can index reference genomes of any size. For genomes less than about 4 billion nucleotides in length, bowtie2-build builds a “small” index using 32-bit numbers in various parts of the index. When the genome is longer, bowtie2-build builds a “large” index using 64-bit numbers. Small indexes are stored in files with the .bt2 extension, and large indexes are stored in files with the .bt2l extension. **The user need not worry about whether a particular index is small or large; the wrapper scripts will automatically build and use the appropriate index.**

However, when I try to build an index from 2000+ genomes without specifying `--large-index`, it fails with the following error:

```
Error: Reference sequence has more than 2^32-1 characters!  Please build a large index by passing the --large-index option to bowtie2-build
Error: Encountered internal Bowtie 2 exception (#1)
Command: […]/conda/91223db3/bin/bowtie2-build-s --wrapper basic-0 -q custom […]/Index/BOWTIE_custom 
```

I understood from the documentation that the `--large-index`option was not mandatory, even if the index is expected to be large. What am I missing here?

Cheers,
Nils

## More information
Install mod: conda
Version 2.3.5

```shell
$ bowtie2-build --version
[...]/bin/bowtie2-build-s version 2.3.5
64-bit
Built on default-9ec70d2b-0cae-4ef9-8b8a-41ee9710f4ef
Mon Apr  1 13:38:55 UTC 2019
Compiler: gcc version 7.3.0 (crosstool-NG 1.23.0.449-a04d0) 
Options: -O3 -m64 -msse2 -funroll-loops -g3 -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -pipe -I[…]/.miniconda3/envs/GriD/include -fdebug-prefix-map=/opt/conda/conda-bld/bowtie2_1554125862557/work=/usr/local/src/conda/bowtie2-2.3.5 -fdebug-prefix-map=/sandbox/users/giordano-n/.miniconda3/envs/GriD=/usr/local/src/conda-prefix -std=c++98 -DPOPCNT_CAPABILITY -DWITH_TBB -DNO_SPINLOCK -DWITH_QUEUELOCK=1
Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}
```

**Comment by ch4rr0** [comment]
Care to share the size of your genome? I tried the command and it seems to be working as expected. I added a print statement to output the size of the reference used to build the index.

```
$ /bowtie2/bowtie2-build lv.fa /tmp/out
lv.fa
4696420496
Settings:
  Output files: "/tmp/out.*.bt2l"
  Line rate: 7 (line is 128 bytes)
  Lines per side: 1 (side is 128 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  lv.fa
Building a LARGE index
Reading reference sizes
...
```

**Comment by nigiord** [comment]
Hi @ch4rr0 ,

Sorry for the delay. The reference file is 12437197918 characters long (12 GB as uncompressed fasta). But I just made some tests, and apparently the problem arises only when I use the `-q` option.

```shell
$bowtie2-build --help
[…]
    -q/--quiet              verbose output (for debugging)
[…]
```

```shell
$bowtie2-build all.fasta BOWTIE_test -q
Error: Reference sequence has more than 2^32-1 characters!  Please build a large index by passing the --large-index option to bowtie2-build
Error: Encountered internal Bowtie 2 exception (#1)
Command: /sandbox/users/giordano-n/.miniconda3/envs/Bowtie2test/bin/bowtie2-build-s --wrapper basic-0 -q all.fasta BOWTIE_test 
Deleting "BOWTIE_test.3.bt2" file written during aborted indexing attempt.
Deleting "BOWTIE_test.4.bt2" file written during aborted indexing attempt.
```
Without this option, everything's fine and a large index is built.

```
$ bowtie2-build all.fasta BOWTIE_test
Settings:
  Output files: "BOWTIE_test.*.bt2l"
  Line rate: 7 (line is 128 bytes)
  Lines per side: 1 (side is 128 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  all.fasta
Building a LARGE index
Reading reference sizes
```

Somehow the `--quiet` option bypass the index size check and default to 32 bits.

Cheers,
Nils

**Comment by ch4rr0** [comment]
Pushed a change to `bug_fixes` branch that should resolve this.

**Comment by ch4rr0** [comment]
This change is available in v2.4.0 and later.
